app/src/main/java/com/example/restrate/auth/LoginFragment.java

```diff
@@ -0,0 +1,132 @@
+    private View view;
+    private TextInputLayout emailET;
+    private TextInputLayout passwordET;
+    private TextView newUser;
+    private Button loginBtn;
+    private ProgressBar loginPB;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        view = inflater.inflate(R.layout.fragment_login, container, false);
+
+        getActivity().getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE);
+
+        ((DrawerLocker) getActivity()).setDrawerEnabled(false);
+
+        emailET = view.findViewById(R.id.login_email);
+        passwordET = view.findViewById(R.id.login_password);
+        loginBtn = view.findViewById(R.id.login_login_btn);
+        newUser = view.findViewById(R.id.login_newUser);
+        loginPB = view.findViewById(R.id.login_pb);
+
+        loginPB.setVisibility(View.INVISIBLE);
+
+        loginBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                login();
+            }
+        });
+
+        newUser.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                navigateToRegister();
+            }
+        });
+
+        return view;
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        if (Model.instance.isUserLoggedIn()) {
+            Log.d("TAG", "User is already logged in: " + Model.instance.getCurrentUser());
+            navigateAfterLoggedIn();
+        }
+    }
+
+    private void login() {
+        loginPB.setVisibility(View.VISIBLE);
+        String email = emailET.getEditText().getText().toString();
+        String password = passwordET.getEditText().getText().toString();
+
+        if (verifyLoginCredentials(email, password)) {
+            Model.instance.login(email, password, new GenericEventListenerWithNoParam() {
+                @Override
+                public void onComplete() {
+                    navigateAfterLoggedIn();
+                    hideKeyboard(getActivity());
+                    loginPB.setVisibility(View.INVISIBLE);
+                }
+            }, new GenericEventListenerWithNoParam() {
+                @Override
+                public void onComplete() {
+                    Toast.makeText(getActivity(), "Email/Password are incorrect", Toast.LENGTH_LONG).show();
+                    loginPB.setVisibility(View.INVISIBLE);
+                }
+            });
+        } else {
+            loginPB.setVisibility(View.INVISIBLE);
+        }
+    }
+
+    private void navigateAfterLoggedIn() {
+//        ((DrawerLocker) getActivity()).setDrawerEnabled(true);
+        Navigation.findNavController(view).navigate(R.id.action_loginFragment_to_restaurantListFragment);
```

Comment:
remove the comment

maybe there is another way to navigate because this has a bug(when using return after login)